tools/deps/prebuilt/include/dispatch/queue_internal.h

```diff
@@ -32,45 +32,63 @@
 #include <dispatch/base.h> // for HeaderDoc
 #endif
 
-#define DISPATCH_OBJ_ASYNC_BIT 0x1
-#define DISPATCH_OBJ_BARRIER_BIT 0x2
-#define DISPATCH_OBJ_GROUP_BIT 0x4
+#define DISPATCH_OBJ_ASYNC_BIT	0x1
+#define DISPATCH_OBJ_BARRIER_BIT	0x2
+#define DISPATCH_OBJ_GROUP_BIT	0x4
 // vtables are pointers far away from the low page in memory
-#define DISPATCH_OBJ_IS_VTABLE(x) ((uintptr_t)(x)->do_vtable > 127ul)
+#define DISPATCH_OBJ_IS_VTABLE(x)	((uintptr_t)(x)->do_vtable > 127ul)
 
 struct dispatch_queue_vtable_s {
-    DISPATCH_VTABLE_HEADER(dispatch_queue_s);
+	DISPATCH_VTABLE_HEADER(dispatch_queue_s);
 };
 
 extern const struct dispatch_queue_vtable_s _dispatch_queue_vtable;
 
-#define DISPATCH_QUEUE_MIN_LABEL_SIZE 64
+#define DISPATCH_QUEUE_MIN_LABEL_SIZE	64
+
+DISPATCH_DECL(dispatch_queue_specific_list);
+DISPATCH_DECL(dispatch_queue_specific);
 
 #ifndef DISPATCH_NO_LEGACY
-#define DISPATCH_QUEUE_HEADER                         \
-    intptr_t dq_running;                              \
-    intptr_t dq_width;                                \
-    struct dispatch_object_s* dq_items_tail;          \
-    struct dispatch_object_s* volatile dq_items_head; \
-    intptr_t dq_serialnum;                            \
-    void* dq_finalizer_ctxt;                          \
-    dispatch_queue_finalizer_function_t dq_finalizer_func
+#define DISPATCH_QUEUE_HEADER \
+	intptr_t dq_running; \
+	intptr_t dq_width; \
+	struct dispatch_object_s *dq_items_tail; \
+	struct dispatch_object_s *volatile dq_items_head; \
+	intptr_t dq_serialnum; \
+	void *dq_finalizer_ctxt; \
+	struct dispatch_queue_s *dq_specific_q; \
+	struct dispatch_queue_specific_list_s *dq_specific_list; \
+	dispatch_queue_finalizer_function_t dq_finalizer_func
 #else
-#define DISPATCH_QUEUE_HEADER                         \
-    intptr_t dq_running;                              \
-    intptr_t dq_width;                                \
-    struct dispatch_object_s* dq_items_tail;          \
-    struct dispatch_object_s* volatile dq_items_head; \
-    intptr_t dq_serialnum;                            \
-    void* dq_finalizer_ctxt;                          \
-    void* dq_manually_drained;                        \
-    bool dq_is_manually_draining
+#define DISPATCH_QUEUE_HEADER \
+	intptr_t dq_running; \
+	intptr_t dq_width; \
+	struct dispatch_object_s *dq_items_tail; \
+	struct dispatch_object_s *volatile dq_items_head; \
+	intptr_t dq_serialnum; \
+	void *dq_finalizer_ctxt; \
+	struct dispatch_queue_s *dq_specific_q; \
+	struct dispatch_queue_specific_list_s *dq_specific_list; \
+	void* dq_manually_drained; \
+	bool dq_is_manually_draining
 #endif
 
+struct dispatch_queue_specific_list_s {
+	TAILQ_HEAD(dispatch_queue_specific_head_s, dispatch_queue_specific_s) contextList;
```

Comment:
Please stick to the naming conventions in this file, so this would be context_list